src/silx/resources/__init__.py

```diff
@@ -237,9 +237,16 @@ def _get_resource_filename(package: str, resource: str) -> str:
     :return: Abolute resource path in the file system
     """
     # Caching prevents extracting the resource twice
-    file_context = importlib_resources.as_file(
-        importlib_resources.files(package) / resource
-    )
+    traversable = importlib_resources.files(package).joinpath(resource)
+    if not traversable.is_file() and not traversable.is_dir():
+        package_dir_context = importlib_resources.as_file(
+            importlib_resources.files(package)
+        )
+        path = _file_manager.enter_context(package_dir_context)
```

Comment:
This needs to be reworked to avoid extracting all resources for a non-existing file path..